src/backends/mysql/blob.cpp

```diff
@@ -18,43 +21,99 @@
 using namespace soci;
 using namespace soci::details;
 
-mysql_blob_backend::mysql_blob_backend(mysql_session_backend &session)
-    : session_(session)
+mysql_blob_backend::mysql_blob_backend(mysql_session_backend &)
+    : details::trivial_blob_backend()
 {
-    throw soci_error("BLOBs are not supported.");
 }
 
 mysql_blob_backend::~mysql_blob_backend()
 {
 }
 
-std::size_t mysql_blob_backend::get_len()
+static unsigned char decode_hex_digit(char c)
 {
-    throw soci_error("BLOBs are not supported.");
+    unsigned char i = static_cast<unsigned char>(tolower(c));
+
+    assert((i >= '0' && i <= '9') || (i >= 'a' && i <= 'f'));
+
+    if (i <= '9') {
+        return i - '0';
+    } else {
+        return i - 'a';
+    }
```

Comment:
This is minor compared to everything else, but it would be nice to respect the SOCI indentation style, i.e. write this as
```suggestion
    if (i <= '9')
    {
        return i - '0';
    }
    else
    {
        return i - 'a';
    }
```

Reply:
Ah yeah - I tried really hard to follow the style. But I guess a few times it slipped through...
If you were using e.g. clang-format for the project, this would be a LOT easier.